src/api/reflector.rs

```diff
@@ -223,3 +224,32 @@ impl<K> Reflector<K> where
         Ok(())
     }
 }
+
+/// ObjectId represents an object by name and namespace (if any)
+#[derive(Ord, PartialOrd, Hash, Eq, PartialEq, Clone)]
+struct ObjectId {
+    name: String,
+    namespace: Option<String>,
+}
+
+impl ToString for ObjectId {
+    fn to_string(&self) -> String {
+        match &self.namespace {
+            Some(ns) => {
+                format!("{} [{}]", self.name, ns)
+            },
+            None => {
+                self.name.clone()
+            }
+        }
+    }
+}
+
+impl From<&ObjectMeta> for ObjectId {
+    fn from(object_meta: &ObjectMeta) -> Self {
+        ObjectId {
+            name: object_meta.name.clone(),
+            namespace: object_meta.namespace.clone(),
+        }
+    }
+}
```

Comment:
This is a nice place to hide this complexity. Nice use of traits on `ObjectId` :+1: